### PR TITLE
Test/soden funcional tests

### PR DESCRIPTION
**Pull request by SodenNedos**

### CODE REVIEW

File: BusinessCore.Tests/FunctionalProgramingTestsSoden.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        {
+            IEnumerable<IEngine> engines = this.vehicleStorage
+                .getAll()
+                /**/
+                .Select(vehicle => vehicle.Engine);
+            Assert.AreEqual(2, engines.Count());
+        }
+
+        [TestMethod]
+        public void c_adds_the_pressure_of_the_wheels_of_the_cars_with_enrollment_number_above_100_and_black_color()
```
Comment (hossmi) [inline BusinessCore.Tests/FunctionalProgramingTestsSoden.cs]: Rename to *the_sum_of_pressures_of_black_vehicles_with_enrollment_number_adobe_to_100_is_6*.

File: BusinessCore.Tests/FunctionalProgramingTestsSoden.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            double pressure = this.vehicleStorage
+               .getAll()
+               /**/
+               .Count();
+
+                Assert.AreEqual(6, pressure);
+        }
+
+        [TestMethod]
+        public void d_from_the_two_white_cars_with_any_doors_get_serial_enrollment_and_horsePower_adove_500()
```
Comment (hossmi) [inline BusinessCore.Tests/FunctionalProgramingTestsSoden.cs]: Rename to *get_serial_enrollment_from_white_vehicles_with_at_least_one_door_and_horsePower_adobe_500cv*.

File: BusinessCore.Tests/FunctionalProgramingTestsSoden.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        {
+            var vehicles = this.vehicleStorage
+                .getAll()
+                /**/
+                .ToArray();
+
+            Assert.AreEqual(1, vehicles.Length);
```
Comment (hossmi) [inline BusinessCore.Tests/FunctionalProgramingTestsSoden.cs]: Add an assert to check the right or expected serial enrollment.

File: BusinessCore.Tests/FunctionalProgramingTestsSoden.cs (hunk 1 above)
Comment (SodenNedos) [reply BusinessCore.Tests/FunctionalProgramingTestsSoden.cs]: Completed.

File: BusinessCore.Tests/FunctionalProgramingTestsSoden.cs (hunk 2 above)
Comment (SodenNedos) [reply BusinessCore.Tests/FunctionalProgramingTestsSoden.cs]: Completed.

**Pull request merged**